bsdbpf/bsd_bpf_sniffer.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2012 Google, Inc. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file in the root of the source
+// tree.
+
+// +build darwin dragonfly freebsd netbsd openbsd
+
+package bsdbpf
+
+import (
+	"github.com/google/gopacket"
+	"golang.org/x/sys/unix"
+
+	"fmt"
+	"syscall"
+	"time"
+	"unsafe"
+)
+
+const wordSize = int(unsafe.Sizeof(uintptr(0)))
+
+func bpfWordAlign(x int) int {
+	return (((x) + (wordSize - 1)) &^ (wordSize - 1))
+}
+
+type Options struct {
+	bpfDeviceName    string
+	readBufLen       int
+	timeout          *syscall.Timeval
+	promisc          bool
+	immediate        bool
+	preserveLinkAddr bool
+}
+
+var defaultOptions = Options{
+	bpfDeviceName:    "",
+	readBufLen:       0,
+	timeout:          nil,
+	promisc:          true,
+	immediate:        false,
+	preserveLinkAddr: true,
+}
+
+type BPFSniffer struct {
```

Comment:
Please add comments to all exported functions and structs, so appropriate godoc can be generated.  Comments should follow http://blog.golang.org/godoc-documenting-go-code format; they should start with the name of the exported object, like so, then describe what it does, what its arguments do, etc:

    // ExportedName foos the bar.
    func ExportedName(...) { }